index.js

```diff
@@ -45,27 +42,29 @@ function getJson (res) {
 // This will allow usage with all Spotify URLs I'm aware of - play., open.
 // and spotify: - but I recommend the latter because it won't contain any
 // weird characters, hopefully.
+
 function getApiUrl (str) {
   if (/(?:album|artist|track)s?[:/][A-Za-z0-9]{22}/.test(str)) {
-    return str.replace(/^.*(artist|album|track)s?[:/]([A-Za-z0-9]{22})$/, (p0, p1, p2) => {
-      return `https://api.spotify.com/v1/${p1}s/${p2}`
-    })
+      return `https://open.spotify.com/oembed?url=` + str;
   } else {
-    return false
+    console.error(str + " does not look like a valid spotify url, so won't be downloaded.");
+    return false;
   }
 }
 
-function startFetching (url) {
-  var apiUrl = getApiUrl(url)
+function startFetching (urls) {
+  apiUrls = [];
 
-  // Exit with error message if the URL is invalid
-  if (!apiUrl) {
-    console.log(`'${url}' is not a valid Spotify URL`)
-    return
-  }
+  urls.forEach((url) => {
+    if(getApiUrl(url)){
+      apiUrls.push(getApiUrl(url));
+    }
+  })
 
-  // Commence callback hell
-  https.get(apiUrl, getJson)
+  for(x in apiUrls){
+    // Commence callback hell
+    https.get(apiUrls[x], getJson);
+  }
```

Comment:
Not a huge fan of building arrays imperatively like this. Original idea was that `startFetching` would be used for individual urls (feels like it would make it more versatile if wanting to extend the program), although I'd be open to doing it differently. How did your thoughts go around this? 😊